plugins/funind/g_indfun.mlg

```diff
@@ -193,18 +193,32 @@ let is_interactive recsl =
   | Vernacextend.VtStartProof _ -> true
   | _ -> false
 
+(* Temporarily switch off some flags during the call to f(). The flags
+   turned off are the ones that should not be triggered by Function
+   when generating the function. *)
+(* TODO 1: make Function not raise non-recursive (call the Definition
+           mechanisme instead)
+   TODO 2: Only for automatically generated names should
+           unused-pattern-matching-variable be turned off. *)
+let with_no_warn (f:unit -> 'a) =
+  let open CWarnings in
+  let s = get_flags () in
+  let _ = set_flags (s^",-matching-variable,-non-recursive") in
+  let res = f() in
```

Comment:
This is wrong if f raises an exception.

Reply:
true. I will fix that.

Reply:
BTW, there have been talks recently of making it possible to locally disable a warning using an attribute (https://coq.zulipchat.com/#narrow/stream/237656-Coq-devs.20.26.20plugin.20devs/topic/Warning.20config.20as.20attribute). So it could make sense to turn this code into a generic function in `cWarnings` (that would be reused for the warning disabling mechanism when it gets implemented).

Reply:
Actually it seems to work anyway, probably because warnings are synched options? Anyway I will usr a proper `unwind/finally` pattern.

Reply:
ok